rdr_service/dao/genomics_dao.py

```diff
@@ -1131,14 +1131,17 @@ def get_id(self, obj):
 
     def get_last_successful_runtime(self, job_id):
         with self.session() as session:
-            return self._get_last_runtime_with_session(session, job_id)
-
-    @staticmethod
-    def _get_last_runtime_with_session(session, job_id):
-        return session.query(functions.max(GenomicJobRun.startTime)) \
-            .filter(GenomicJobRun.jobId == job_id,
-                    GenomicJobRun.runResult == GenomicSubProcessResult.SUCCESS) \
-            .one()[0]
+            return session.query(
+                functions.max(GenomicJobRun.startTime)
+            ).filter(GenomicJobRun.jobId == job_id,
+                     GenomicJobRun.runStatus == GenomicSubProcessStatus.COMPLETED,
+                     and_(
+                         or_(
+                             GenomicJobRun.runResult == GenomicSubProcessResult.SUCCESS,
+                             GenomicJobRun.runResult == GenomicSubProcessResult.NO_FILES
+                         )
+                     )
```

Comment:
Could this be simplified to the following?:
`GenomicJobRun.runStatus == GenomicSubProcessStatus.COMPLETED,
GenomicJobRun.runResult.in_(GenomicSubProcessResult.SUCCESS, GenomicSubProcessResult.NO_FILES)`
In MySql, IN is faster than OR statements.

Reply:
https://github.com/all-of-us/raw-data-repository/pull/2897/commits/18961eb43d857fcffaba5238da33f6afb4a37e09